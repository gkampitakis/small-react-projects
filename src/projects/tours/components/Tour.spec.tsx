export {};

describe('Tour', () => {
  it('Should render correctly', () => {
    const data = {
      id: '1',
      image: 'string',
      name: 'mockName',
      info: 'data',
      price: '12345'
    };
    const removeTour = jest.fn();
  });

  it('Should trim info string if bigger than 200 chars', () => {
    const data = {
      id: '1',
      image: 'string',
      name: 'mockName',
      info: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla ac blandit augue, in auctor dolor. 
      Integer tempor, libero eu pellentesque rutrum, lorem ante varius elit, eu aliquam magna massa sit amet metus. In hac habitasse platea dictumst. Ut purus orci, sollicitudin et efficitur pretium, viverra suscipit velit. 
      Ut vehicula vulputate ex, aliquam venenatis felis.`,
      price: '12345'
    };
    const removeTour = jest.fn();
  });

  it('Should invoke passed function on button click', () => {
    const data = {
      id: '1',
      image: 'string',
      name: 'mockName',
      info: 'data',
      price: '12345'
    };
    const removeTour = jest.fn();
  });
});
