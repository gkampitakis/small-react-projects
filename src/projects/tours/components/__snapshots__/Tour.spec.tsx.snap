// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tour Should render correctly 1`] = `
<body>
  <div>
    <article
      class="single-tour"
    >
      <img
        alt="mockName"
        src="string"
      />
      <footer>
        <div
          class="tour-info"
        >
          <h4>
            mockName
          </h4>
          <h4
            class="tour-price"
          >
            12345
          </h4>
        </div>
        <p>
          data...
          <button>
            read more
          </button>
        </p>
        <button
          class="delete-btn"
        >
          Not interested
        </button>
      </footer>
    </article>
  </div>
</body>
`;
